<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>plot GInteractions — loopBouquetPlot • geomeTriD</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="plot GInteractions — loopBouquetPlot"><meta property="og:description" content="plot graph for GInteractions"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">geomeTriD</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/geomeTriD.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jianhong/geometrid" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>plot GInteractions</h1>
    <small class="dont-index">Source: <a href="https://github.com/jianhong/geomeTriD/blob/main/R/loopBouquetPlot.R" class="external-link"><code>R/loopBouquetPlot.R</code></a></small>
    <div class="hidden name"><code>loopBouquetPlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>plot graph for GInteractions</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">loopBouquetPlot</span><span class="op">(</span></span>
<span>  <span class="va">gi</span>,</span>
<span>  <span class="va">range</span>,</span>
<span>  <span class="va">feature.gr</span>,</span>
<span>  <span class="va">genomicSigs</span>,</span>
<span>  signalTransformFun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="va">x</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span> <span class="op">}</span>,</span>
<span>  label_region <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  show_edges <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_cluster <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  lwd.backbone <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  col.backbone <span class="op">=</span> <span class="st">"gray"</span>,</span>
<span>  lwd.maxGenomicSigs <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  reverseGenomicSigs <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  col.backbone_background <span class="op">=</span> <span class="st">"gray70"</span>,</span>
<span>  alpha.backbone_background <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  lwd.gene <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  lwd.nodeCircle <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  col.nodeCircle <span class="op">=</span> <span class="st">"#DDDDDD25"</span>,</span>
<span>  lwd.edge <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  col.edge <span class="op">=</span> <span class="st">"gray80"</span>,</span>
<span>  coor_mark_interval <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  col.coor <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  show_coor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  coor_tick_unit <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  label_gene <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  col.tension_line <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>  lwd.tension_line <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  length.arrow <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  safe_text_force <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  method <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  doReduce <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gi">gi<a class="anchor" aria-label="anchor" href="#arg-gi"></a></dt>
<dd><p>An object of <a href="https://rdrr.io/pkg/InteractionSet/man/GInteractions-class.html" class="external-link">GInteractions</a></p></dd>


<dt id="arg-range">range<a class="anchor" aria-label="anchor" href="#arg-range"></a></dt>
<dd><p>The region to plot. an object of <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></p></dd>


<dt id="arg-feature-gr">feature.gr<a class="anchor" aria-label="anchor" href="#arg-feature-gr"></a></dt>
<dd><p>The annotation features to be added. An object of <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a>.</p></dd>


<dt id="arg-genomicsigs">genomicSigs<a class="anchor" aria-label="anchor" href="#arg-genomicsigs"></a></dt>
<dd><p>The genomic signals. An object of <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> with scores or an object of <a href="https://rdrr.io/pkg/trackViewer/man/trackStyle-class.html" class="external-link">track</a>.</p></dd>


<dt id="arg-signaltransformfun">signalTransformFun<a class="anchor" aria-label="anchor" href="#arg-signaltransformfun"></a></dt>
<dd><p>The transformation function for genomic signals.</p></dd>


<dt id="arg-label-region">label_region<a class="anchor" aria-label="anchor" href="#arg-label-region"></a></dt>
<dd><p>Label the region node or not.</p></dd>


<dt id="arg-show-edges">show_edges<a class="anchor" aria-label="anchor" href="#arg-show-edges"></a></dt>
<dd><p>Plot the interaction edges or not.</p></dd>


<dt id="arg-show-cluster">show_cluster<a class="anchor" aria-label="anchor" href="#arg-show-cluster"></a></dt>
<dd><p>Plot the cluster background or not.</p></dd>


<dt id="arg-lwd-backbone-lwd-gene-lwd-nodecircle-lwd-edge-lwd-tension-line-lwd-maxgenomicsigs">lwd.backbone, lwd.gene, lwd.nodeCircle, lwd.edge, lwd.tension_line, lwd.maxGenomicSigs<a class="anchor" aria-label="anchor" href="#arg-lwd-backbone-lwd-gene-lwd-nodecircle-lwd-edge-lwd-tension-line-lwd-maxgenomicsigs"></a></dt>
<dd><p>Line width for the
linker, gene, interaction node circle, the dashed line of interaction edges, the tension line and the maximal reversed genomic signal.</p></dd>


<dt id="arg-col-backbone-col-backbone-background-col-nodecircle-col-edge-col-tension-line-col-coor">col.backbone, col.backbone_background, col.nodeCircle, col.edge, col.tension_line, col.coor<a class="anchor" aria-label="anchor" href="#arg-col-backbone-col-backbone-background-col-nodecircle-col-edge-col-tension-line-col-coor"></a></dt>
<dd><p>Color
for the DNA chain, the compact DNA chain, the node circle, the linker, the tension line and the coordinates marker.</p></dd>


<dt id="arg-reversegenomicsigs">reverseGenomicSigs<a class="anchor" aria-label="anchor" href="#arg-reversegenomicsigs"></a></dt>
<dd><p>Plot the Genomic signals in reverse values.</p></dd>


<dt id="arg-alpha-backbone-background">alpha.backbone_background<a class="anchor" aria-label="anchor" href="#arg-alpha-backbone-background"></a></dt>
<dd><p>Alpha channel for transparency of backbone background.</p></dd>


<dt id="arg-coor-mark-interval">coor_mark_interval<a class="anchor" aria-label="anchor" href="#arg-coor-mark-interval"></a></dt>
<dd><p>The coordinates marker interval. Numeric(1). Set to 0
to turn it off. The default value 1e5 means show coordinates every 0.1M bp.</p></dd>


<dt id="arg-show-coor">show_coor<a class="anchor" aria-label="anchor" href="#arg-show-coor"></a></dt>
<dd><p>Show coordinates or not.</p></dd>


<dt id="arg-coor-tick-unit">coor_tick_unit<a class="anchor" aria-label="anchor" href="#arg-coor-tick-unit"></a></dt>
<dd><p>The bps for every ticks. Default is 1K.</p></dd>


<dt id="arg-label-gene">label_gene<a class="anchor" aria-label="anchor" href="#arg-label-gene"></a></dt>
<dd><p>Show gene symbol or not.</p></dd>


<dt id="arg-length-arrow">length.arrow<a class="anchor" aria-label="anchor" href="#arg-length-arrow"></a></dt>
<dd><p>Length of the edges of the arrow head (in inches).</p></dd>


<dt id="arg-safe-text-force">safe_text_force<a class="anchor" aria-label="anchor" href="#arg-safe-text-force"></a></dt>
<dd><p>The loops to avoid the text overlapping.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>Plot method. Could be 1 or 2.</p></dd>


<dt id="arg-doreduce">doReduce<a class="anchor" aria-label="anchor" href="#arg-doreduce"></a></dt>
<dd><p>Reduce the GInteractions or not.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Parameter will be passed to <a href="https://r.igraph.org/reference/layout_with_fr.html" class="external-link">layout_with_fr</a>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A invisible list with the key points of the plot.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">InteractionSet</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'MatrixGenerics'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:matrixStats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'Biobase'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:MatrixGenerics':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from 'package:matrixStats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-in"><span><span class="va">gi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"gi.rds"</span>, package <span class="op">=</span> <span class="st">"trackViewer"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">range</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"chr2"</span>, <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fl">234500000</span>, <span class="fl">235000000</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   403 genes were dropped because they have exons located on both strands</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   of the same reference sequence or on more than one reference sequence,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   so cannot be represented by a single genomic range.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Use 'single.strand.genes.only=FALSE' to get all the genes in a</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   GRangesList object, or use suppressMessages() to suppress this message.</span>
<span class="r-in"><span><span class="va">feature.gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">subsetByOverlaps</a></span><span class="op">(</span><span class="va">feature.gr</span>, <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/InteractionSet/man/interaction-accessors.html" class="external-link">regions</a></span><span class="op">(</span><span class="va">gi</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">symbols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">mget</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">$</span><span class="va">gene_id</span>, <span class="va">org.Hs.egSYMBOL</span>, ifnotfound <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">feature.gr</span><span class="op">$</span><span class="va">label</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">feature.gr</span><span class="op">$</span><span class="va">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">feature.gr</span><span class="op">$</span><span class="va">type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cRE"</span>, <span class="st">"gene"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  replace <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.9</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">feature.gr</span><span class="op">$</span><span class="va">pch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">feature.gr</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">feature.gr</span><span class="op">$</span><span class="va">pch</span><span class="op">[</span><span class="va">feature.gr</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"cRE"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">11</span></span></span>
<span class="r-in"><span><span class="fu">loopBouquetPlot</span><span class="op">(</span><span class="va">gi</span>, <span class="va">range</span>, <span class="va">feature.gr</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="loopBouquetPlot-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

